scripts/quests/windurst/Toraimarai_Turmoil.lua

```diff
@@ -81,9 +76,38 @@ quest.sections =
                         player:confirmTrade()
                         -- From previous implementation, award 100 fame (50 + 50) on first completion,
                         -- and 50 fame for any subsequent trade.
-                        if player:getQuestStatus(quest.areaId, quest.questId) == QUEST_ACCEPTED then
-                            player:addFame(xi.quest.fame_area.WINDURST, 50)
-                        end
+                        player:addFame(xi.quest.fame_area.WINDURST, 50)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        --repeat completion
+        check = function(player, status, vars)
+            return status == QUEST_COMPLETED
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ohbiru-Dohbiru'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(795, 4500, 0, xi.items.STARMITE_SHELL) --dialog for repeat
+                end,
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { { xi.items.STARMITE_SHELL, 3 } }) then
+                        return quest:progressEvent(791)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [791] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
```

Comment:
Now it just needs the addFame(50) in here, and the change above to 100 fame, and we're good I think.